### PR TITLE
[docs] Clarify the position regarding bot generated and AI-assisted PRs

### PR DESCRIPTION
**Pull request by reyang**
There were cases in other OpenTelemetry repositories, and it happened in OpenTelemetry .NET recently (e.g. #5893).
Trying to put some policy here.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -203,13 +203,18 @@ Open a pull request against the main `opentelemetry-dotnet` repo.
...
+* Submit small, focused PRs addressing a single concern/issue.
 * Make sure the PR title reflects the contribution.
 * Write a summary that helps understand the change.
 * Include usage examples in the summary, where applicable.
 * Include benchmarks (before/after) in the summary, for contributions that are
   performance enhancements.
+* We are open to bot generated PRs or AI/LLM assisted PRs. Actually, we are
+  using
+  [dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/configuring-dependabot-security-updates)
+  to automate the security updates. However, if you use bots to generate spams,
```
Comment (cijothomas) [inline CONTRIBUTING.md:215]: spam PRs will anyway be closed, whether bot generated or human generated.

Comment (reyang) [reply CONTRIBUTING.md:215]: I think there is a difference - we don't normally block human accounts unless there is a clear violation to the code of conduct.

Comment (CodeBlanch) [inline CONTRIBUTING.md:215]: This language `if you use bots to generate spams,` seems a little odd.

Probably at the least "spams" should be "spam PRs" or "spammy PRs"? Maybe a bit of definition would also help?

`if you use bots to generate spam PRs (eg incorrect, noisy, non-improvements, unintelligible, etc.),`

Comment (reyang) [reply CONTRIBUTING.md]: Good point, I've tweaked the wording.

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by TimothyMothra — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Pull request merged**